tests/prague/eip7685_general_purpose_el_requests/test_deposits_withdrawals_consolidations.py

```diff
@@ -0,0 +1,376 @@
+def test_valid_deposit_withdrawal_consolidation_request_from_same_tx(
+    blockchain_test: BlockchainTestFiller,
+    pre: Alloc,
+    requests: List[DepositRequest | WithdrawalRequest | ConsolidationRequest],
+):
+    """
+    Test making a deposit to the beacon chain deposit contract and a withdrawal in the same tx.
+    """
+    withdrawal_request_fee = 1
+    consolidation_request_fee = 1
+
+    calldata = b""
+    contract_code = Bytecode()
+    total_value = 0
+    storage = Storage()
+
+    for request in requests:
+        calldata_start = len(calldata)
+        current_calldata = request.calldata
+        calldata += current_calldata
+
+        contract_code += Op.CALLDATACOPY(0, calldata_start, len(current_calldata))
+
+        call_contract_address = 0
+        value = 0
+        if isinstance(request, DepositRequest):
+            call_contract_address = Spec_EIP6110.DEPOSIT_CONTRACT_ADDRESS
+            value = request.value
+        elif isinstance(request, WithdrawalRequest):
+            call_contract_address = Spec_EIP7002.WITHDRAWAL_REQUEST_PREDEPLOY_ADDRESS
+            value = withdrawal_request_fee
+        elif isinstance(request, ConsolidationRequest):
+            call_contract_address = Spec_EIP7251.CONSOLIDATION_REQUEST_PREDEPLOY_ADDRESS
+            value = consolidation_request_fee
+
+        total_value += value
+
+        contract_code += Op.SSTORE(
+            storage.store_next(1),
+            Op.CALL(
+                address=call_contract_address,
+                value=value,
+                args_offset=0,
+                args_size=len(current_calldata),
+            ),
+        )
+
+    sender = pre.fund_eoa()
+    contract_address = pre.deploy_contract(
+        code=contract_code,
+    )
+
+    tx = Transaction(
+        gas_limit=10_000_000,
+        to=contract_address,
+        value=total_value,
+        data=calldata,
+        sender=sender,
+    )
+
+    blockchain_test(
+        genesis_environment=Environment(),
+        pre=pre,
+        post={
+            contract_address: Account(
+                storage=storage,
+            )
+        },
+        blocks=[
+            Block(
+                txs=[tx],
+                header_verify=Header(
+                    requests_root=[
+                        request.with_source_address(contract_address)
+                        for request in sorted(requests, key=lambda r: r.type_byte())
+                    ]
+                ),
+            )
+        ],
+    )
+
+
+@pytest.mark.parametrize(
+    "requests,block_body_override_requests,exception",
+    [
+        pytest.param(
+            [
+                single_withdrawal_from_eoa(0),
+                single_deposit_from_eoa(0),
+            ],
+            [
+                single_withdrawal(0).with_source_address(TestAddress),
+                single_deposit(0),
+            ],
+            # TODO: on the Engine API, the issue should be detected as an invalid block hash
```

Comment:
Yeah, let's leave it to the PR that actually checks for the exceptions, I can fix it then since in that PR we have the infra ready to check for exceptions.